File: internal/model/vo/category.go
package vo

// CategoryVO 分类响应
type CategoryVO struct {
	CategoryID uint   `json:"category_id"`
	UserID     uint   `json:"user_id"`
	Name       string `json:"name"`
	Color      string `json:"color"`
	Icon       string `json:"icon"`
	SortOrder  int    `json:"sort_order"`
	CreatedAt  string `json:"created_at"`
}

// CategoryListVO 分类列表响应
type CategoryListVO struct {
	List  []CategoryVO `json:"list"`
	Total int64        `json:"total"`
	Page  int          `json:"page"`
}

File: src/model/vo/category.ts
export interface CategoryVO {
  category_id: number;
  user_id: number;
  name: string;
  color: string | null;
  icon: string | null;
  sort_order: number;
  created_at: string;
}

export function toCategoryVO(cat: {
  id: number;
  userId: number;
  name: string;
  color: string | null;
  icon: string | null;
  sortOrder: number;
  createdAt: string;
}): CategoryVO {
  return {
    category_id: cat.id,
    user_id: cat.userId,
    name: cat.name,
    color: cat.color,
    icon: cat.icon,
    sort_order: cat.sortOrder,
    created_at: cat.createdAt,
  };
}
